src/item_factory.cpp

```diff
@@ -134,11 +134,13 @@ void Item_factory::finalize() {
             }
 
             // if the gun doesn't have a DEFAULT mode then add one now
-            obj.gun->modes.emplace( "DEFAULT", std::tuple<std::string, int, std::set<std::string>>( defmode, 1, {} ) );
+            obj.gun->modes.emplace( "DEFAULT", std::tuple<std::string, int, std::set<std::string>>( defmode, 1,
+                                    std::set<std::string>() ) );
```

Comment:
It's mentioned in the compiler's output: `chosen constructor is explicit in copy-initialization`. Old versions of `clang` seem to unable to deal with initializers.